cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -436,22 +436,11 @@ void MapNode::SubmitTask(std::function<Result<ExecBatch>(ExecBatch)> map_fn,
     return Status::OK();
   };
 
-  if (executor_) {
-    status = task_group_.AddTask([this, task]() -> Result<Future<>> {
-      return this->executor_->Submit(this->stop_source_.token(), [this, task]() {
-        auto status = task();
-        if (this->input_counter_.Increment()) {
-          this->Finish(status);
-        }
-        return status;
-      });
-    });
-  } else {
-    status = task();
-    if (input_counter_.Increment()) {
-      this->Finish(status);
-    }
+  status = task();
+  if (input_counter_.Increment()) {
```

Comment:
I think it's slightly clearer to readers if this check comes after the `if (!status.ok())` check.  Then we can just do...

```
if (input_counter_.Increment()) {
  this->Finish();
}
```

Reply:
Made the change.